# Gendering the talks 

One of our areas of interest are to see if TED talks by men and women are different. If there are differences, we would like to see if we can detect the gender of a speaker based on the words in a TED talk transcript. To pursue these avenues, we begin by "gendering" talks when possible. 

In this notebook, we use the genders of the speakers to "gender" the TED talks themselves. A talk with one speaker inherits the gender of the speaker. For a talk with two speakers, if the genders of the speakers are the same, then we proceed as if the talk had one speaker. For talks with two speakers where the genders of the speakers is not the same, we place these talks to the side. 

In [1]:
import pandas as pd
import csv
import string

In [2]:
# Load the gendered speaker file:
speakers = pd.read_csv("speakers_with_gender.csv")

In [3]:
ted_only = pd.read_csv('../data/Release_v0/TEDonly_final.csv')
ted_plus = pd.read_csv('../data/Release_v0/TEDplus_final.csv')

In [4]:
ted_only.head()

Unnamed: 0  Unnamed: 0.1  Talk_ID  \
0           0             0        1   
1           1             1        7   
2           2             2       53   
3           3             3       66   
4           4             4       92   

                                          public_url  \
0  https://www.ted.com/talks/al_gore_on_averting_...   
1  https://www.ted.com/talks/david_pogue_says_sim...   
2  https://www.ted.com/talks/majora_carter_s_tale...   
3  https://www.ted.com/talks/ken_robinson_says_sc...   
4  https://www.ted.com/talks/hans_rosling_shows_t...   

                          headline  \
0      Averting the climate crisis   
1                 Simplicity sells   
2              Greening the ghetto   
3      Do schools kill creativity?   
4  The best stats you've ever seen   

                                         description    event duration  \
0  With the same humor and humanity he exuded in ...  TED2006  0:16:17   
1  New York Times columnist David Pogue takes aim...  TED2006  0:21:26   
2  In an emotionally charged talk, MacArthur-winn...  TED2006  0:18:36   
3  Sir Ken Robinson makes an entertaining and pro...  TED2006  0:19:24   
4  You've never seen data presented like this. Wi...  TED2006  0:19:50   

  published                                               tags     views  \
0   6/27/06  alternative energy,cars,global issues,climate ...   3266733   
1   6/27/06  simplicity,entertainment,interface design,soft...   1702201   
2   6/27/06  MacArthur grant,cities,green,activism,politics...   2000421   
3   6/27/06  children,teaching,creativity,parenting,culture...  51614087   
4   6/27/06  demo,Asia,global issues,visualizations,global ...  12662135   

                                                text      speaker_1 speaker_2  \
0    Thank you so much, Chris. And it's truly a g...        Al Gore       NaN   
1    (Music: "The Sound of Silence," Simon & Garf...    David Pogue       NaN   
2    If you're here today — and I'm very happy th...  Majora Carter       NaN   
3    Good morning. How are you?    (Laughter)    ...   Ken Robinson       NaN   
4    About 10 years ago, I took on the task to te...   Hans Rosling       NaN   

  speaker_3 speaker_4  
0       NaN       NaN  
1       NaN       NaN  
2       NaN       NaN  
3       NaN       NaN  
4       NaN       NaN

In [5]:
# Set the talk ID as the index and drop the unnecessary first two columns: 
ted_only = ted_only.set_index('Talk_ID')
ted_only = ted_only.drop(columns = ['Unnamed: 0', 'Unnamed: 0.1'])

In [6]:
ted_only.head()

public_url  \
Talk_ID                                                      
1        https://www.ted.com/talks/al_gore_on_averting_...   
7        https://www.ted.com/talks/david_pogue_says_sim...   
53       https://www.ted.com/talks/majora_carter_s_tale...   
66       https://www.ted.com/talks/ken_robinson_says_sc...   
92       https://www.ted.com/talks/hans_rosling_shows_t...   

                                headline  \
Talk_ID                                    
1            Averting the climate crisis   
7                       Simplicity sells   
53                   Greening the ghetto   
66           Do schools kill creativity?   
92       The best stats you've ever seen   

                                               description    event duration  \
Talk_ID                                                                        
1        With the same humor and humanity he exuded in ...  TED2006  0:16:17   
7        New York Times columnist David Pogue takes aim...  TED2006  0:21:26   
53       In an emotionally charged talk, MacArthur-winn...  TED2006  0:18:36   
66       Sir Ken Robinson makes an entertaining and pro...  TED2006  0:19:24   
92       You've never seen data presented like this. Wi...  TED2006  0:19:50   

        published                                               tags  \
Talk_ID                                                                
1         6/27/06  alternative energy,cars,global issues,climate ...   
7         6/27/06  simplicity,entertainment,interface design,soft...   
53        6/27/06  MacArthur grant,cities,green,activism,politics...   
66        6/27/06  children,teaching,creativity,parenting,culture...   
92        6/27/06  demo,Asia,global issues,visualizations,global ...   

            views                                               text  \
Talk_ID                                                                
1         3266733    Thank you so much, Chris. And it's truly a g...   
7         1702201    (Music: "The Sound of Silence," Simon & Garf...   
53        2000421    If you're here today — and I'm very happy th...   
66       51614087    Good morning. How are you?    (Laughter)    ...   
92       12662135    About 10 years ago, I took on the task to te...   

             speaker_1 speaker_2 speaker_3 speaker_4  
Talk_ID                                               
1              Al Gore       NaN       NaN       NaN  
7          David Pogue       NaN       NaN       NaN  
53       Majora Carter       NaN       NaN       NaN  
66        Ken Robinson       NaN       NaN       NaN  
92        Hans Rosling       NaN       NaN       NaN

First, we want to know which talks only have one speaker. To do this, we only select the rows that `NaN` for the second speaker. 

In [51]:
talk1speak = ted_only[ted_only['speaker_2'].isnull()]

# https://stackoverflow.com/questions/43831539/how-to-select-rows-with-nan-in-particular-column

In [52]:
talk1speak.shape

(966, 13)

In [72]:
talk1 = talk1speak[["speaker_1","speaker_2","speaker_3","speaker_4"]]

In [74]:
talk1.head()


speaker_1 speaker_2 speaker_3 speaker_4
Talk_ID                                             
1              Al Gore       NaN       NaN       NaN
7          David Pogue       NaN       NaN       NaN
53       Majora Carter       NaN       NaN       NaN
66        Ken Robinson       NaN       NaN       NaN
92        Hans Rosling       NaN       NaN       NaN

We have 966 talks that have only one speaker. 


To figure out which talks have exactly two speakers, we select the rows that have a second speaker, but **not** a third one:  

In [64]:
s2temp = ted_only[ted_only['speaker_3'].isnull()]
talk2speak = s2temp[~s2temp['speaker_2'].isnull()]

In [65]:
talk2speak.shape

(25, 13)

In [75]:
talk2 = talk2speak[["speaker_1","speaker_2","speaker_3","speaker_4"]]

We have 25 talks that have two speakers. 


To figure out which talks have three speakers, we select the rows that have a third speaker, but **not** a fourth one:  

In [11]:
s3temp = ted_only[ted_only['speaker_4'].isnull()]
talk3speak = s3temp[~s3temp['speaker_3'].isnull()]

In [12]:
talk3speak.shape

(0, 13)

There are no talks with exactly three speakers. 


To figure out which talks have four speakers, we select the rows that have a fourth speaker:  

In [13]:
talk4speak = ted_only[~ted_only['speaker_4'].isnull()]


In [14]:
talk4speak.shape

(1, 13)

In [76]:
talk4 = talk4speak[["speaker_1","speaker_2","speaker_3","speaker_4"]]

As a quick check, we check if the total number of rows in the `talkNspeak` data frames equal the number of rows in `ted_only`:

In [15]:
ted_only.shape[0] == talk1speak.shape[0] + talk2speak.shape[0] + talk3speak.shape[0] + talk4speak.shape[0]

True

## Adding gender to the talks

Now that we have separate the talks into smaller data frames, each representing the number of speakers, we will be working to "gender" the talks. 

In [16]:
gender_slice = speakers[["speaker","Gender_handcheck"]]

In [17]:
gender_slice.head()

speaker Gender_handcheck
0        Al Gore             male
1    David Pogue             male
2  Majora Carter           female
3   Ken Robinson             male
4   Hans Rosling             male

In [27]:
gender_slice.set_index("speaker")

Gender_handcheck
speaker                                
Al Gore                            male
David Pogue                        male
Majora Carter                    female
Ken Robinson                       male
Hans Rosling                       male
...                                 ...
Improv Everywhere    they (group mixed)
Quixotic Fusion      they (group mixed)
Silk Road Ensemble   they (group mixed)
The LXD              they (group mixed)
Rodrigo y Gabriela  they (male, female)

[868 rows x 1 columns]

### Gendering talks with one speaker

In [53]:
# We want to take speaker gender from speakers and put it into the talks
talk1speak = talk1speak.reset_index().merge(gender_slice, left_on = "speaker_1", right_on = "speaker", how = "left")

# We use `reset_index()` to preserve the `Talk_ID` column

In [54]:
talk1speak.head()

Talk_ID                                         public_url  \
0        1  https://www.ted.com/talks/al_gore_on_averting_...   
1        7  https://www.ted.com/talks/david_pogue_says_sim...   
2       53  https://www.ted.com/talks/majora_carter_s_tale...   
3       66  https://www.ted.com/talks/ken_robinson_says_sc...   
4       92  https://www.ted.com/talks/hans_rosling_shows_t...   

                          headline  \
0      Averting the climate crisis   
1                 Simplicity sells   
2              Greening the ghetto   
3      Do schools kill creativity?   
4  The best stats you've ever seen   

                                         description    event duration  \
0  With the same humor and humanity he exuded in ...  TED2006  0:16:17   
1  New York Times columnist David Pogue takes aim...  TED2006  0:21:26   
2  In an emotionally charged talk, MacArthur-winn...  TED2006  0:18:36   
3  Sir Ken Robinson makes an entertaining and pro...  TED2006  0:19:24   
4  You've never seen data presented like this. Wi...  TED2006  0:19:50   

  published                                               tags     views  \
0   6/27/06  alternative energy,cars,global issues,climate ...   3266733   
1   6/27/06  simplicity,entertainment,interface design,soft...   1702201   
2   6/27/06  MacArthur grant,cities,green,activism,politics...   2000421   
3   6/27/06  children,teaching,creativity,parenting,culture...  51614087   
4   6/27/06  demo,Asia,global issues,visualizations,global ...  12662135   

                                                text      speaker_1 speaker_2  \
0    Thank you so much, Chris. And it's truly a g...        Al Gore       NaN   
1    (Music: "The Sound of Silence," Simon & Garf...    David Pogue       NaN   
2    If you're here today — and I'm very happy th...  Majora Carter       NaN   
3    Good morning. How are you?    (Laughter)    ...   Ken Robinson       NaN   
4    About 10 years ago, I took on the task to te...   Hans Rosling       NaN   

  speaker_3 speaker_4        speaker Gender_handcheck  
0       NaN       NaN        Al Gore             male  
1       NaN       NaN    David Pogue             male  
2       NaN       NaN  Majora Carter           female  
3       NaN       NaN   Ken Robinson             male  
4       NaN       NaN   Hans Rosling             male

In [55]:
# Reset the index back to `Talk_ID` 
talk1speak.set_index("Talk_ID", inplace = True)

In [56]:
talk1speak

public_url  \
Talk_ID                                                      
1        https://www.ted.com/talks/al_gore_on_averting_...   
7        https://www.ted.com/talks/david_pogue_says_sim...   
53       https://www.ted.com/talks/majora_carter_s_tale...   
66       https://www.ted.com/talks/ken_robinson_says_sc...   
92       https://www.ted.com/talks/hans_rosling_shows_t...   
...                                                    ...   
3623     https://www.ted.com/talks/atul_gawande_want_to...   
8588     https://www.ted.com/talks/jacob_collier_a_one_...   
8606     https://www.ted.com/talks/anna_rosling_ronnlun...   
9987     https://www.ted.com/talks/guy_winch_how_to_fix...   
10807    https://www.ted.com/talks/chuck_nice_a_funny_l...   

                                                  headline  \
Talk_ID                                                      
1                              Averting the climate crisis   
7                                         Simplicity sells   
53                                     Greening the ghetto   
66                             Do schools kill creativity?   
92                         The best stats you've ever seen   
...                                                    ...   
3623           Want to get great at something? Get a coach   
8588                          A one-man musical phenomenon   
8606     See how the rest of the world lives, organized...   
9987                             How to fix a broken heart   
10807    A funny look at the unintended consequences of...   

                                               description    event duration  \
Talk_ID                                                                        
1        With the same humor and humanity he exuded in ...  TED2006  0:16:17   
7        New York Times columnist David Pogue takes aim...  TED2006  0:21:26   
53       In an emotionally charged talk, MacArthur-winn...  TED2006  0:18:36   
66       Sir Ken Robinson makes an entertaining and pro...  TED2006  0:19:24   
92       You've never seen data presented like this. Wi...  TED2006  0:19:50   
...                                                    ...      ...      ...   
3623     How do we improve in the face of complexity? A...  TED2017  0:16:58   
8588     Jacob Collier is a one-man band and force of n...  TED2017  0:15:14   
8606     What does it look like when someone in Sweden ...  TED2017  0:11:41   
9987     At some point in our lives, almost every one o...  TED2017  0:12:25   
10807    Technology should work for us, but what happen...  TED2017  0:10:42   

        published                                               tags  \
Talk_ID                                                                
1         6/27/06  alternative energy,cars,global issues,climate ...   
7         6/27/06  simplicity,entertainment,interface design,soft...   
53        6/27/06  MacArthur grant,cities,green,activism,politics...   
66        6/27/06  children,teaching,creativity,parenting,culture...   
92        6/27/06  demo,Asia,global issues,visualizations,global ...   
...           ...                                                ...   
3623     12/15/17  Surgery,goal-setting,potential,teaching,life,s...   
8588      1/12/18  visualizations,performance art,live music,voca...   
8606      1/18/18  global issues,visualizations,photography,data,...   
9987       2/5/18  love,addiction,depression,compassion,personal ...   
10807     2/27/18  AI,funny,comedy,computers,technology,social me...   

            views                                               text  \
Talk_ID                                                                
1         3266733    Thank you so much, Chris. And it's truly a g...   
7         1702201    (Music: "The Sound of Silence," Simon & Garf...   
53        2000421    If you're here today — and I'm very happy th...   
66       51614087    Good morning. How are you?    (Laughter)    ...   
92       12662135    Abo

In [57]:
# Remove the second speaker column and rename the last column to "talk_gender"
talk1speak = talk1speak.drop(columns = ['speaker'])

In [58]:
talk1speak.head()

public_url  \
Talk_ID                                                      
1        https://www.ted.com/talks/al_gore_on_averting_...   
7        https://www.ted.com/talks/david_pogue_says_sim...   
53       https://www.ted.com/talks/majora_carter_s_tale...   
66       https://www.ted.com/talks/ken_robinson_says_sc...   
92       https://www.ted.com/talks/hans_rosling_shows_t...   

                                headline  \
Talk_ID                                    
1            Averting the climate crisis   
7                       Simplicity sells   
53                   Greening the ghetto   
66           Do schools kill creativity?   
92       The best stats you've ever seen   

                                               description    event duration  \
Talk_ID                                                                        
1        With the same humor and humanity he exuded in ...  TED2006  0:16:17   
7        New York Times columnist David Pogue takes aim...  TED2006  0:21:26   
53       In an emotionally charged talk, MacArthur-winn...  TED2006  0:18:36   
66       Sir Ken Robinson makes an entertaining and pro...  TED2006  0:19:24   
92       You've never seen data presented like this. Wi...  TED2006  0:19:50   

        published                                               tags  \
Talk_ID                                                                
1         6/27/06  alternative energy,cars,global issues,climate ...   
7         6/27/06  simplicity,entertainment,interface design,soft...   
53        6/27/06  MacArthur grant,cities,green,activism,politics...   
66        6/27/06  children,teaching,creativity,parenting,culture...   
92        6/27/06  demo,Asia,global issues,visualizations,global ...   

            views                                               text  \
Talk_ID                                                                
1         3266733    Thank you so much, Chris. And it's truly a g...   
7         1702201    (Music: "The Sound of Silence," Simon & Garf...   
53        2000421    If you're here today — and I'm very happy th...   
66       51614087    Good morning. How are you?    (Laughter)    ...   
92       12662135    About 10 years ago, I took on the task to te...   

             speaker_1 speaker_2 speaker_3 speaker_4 Gender_handcheck  
Talk_ID                                                                
1              Al Gore       NaN       NaN       NaN             male  
7          David Pogue       NaN       NaN       NaN             male  
53       Majora Carter       NaN       NaN       NaN           female  
66        Ken Robinson       NaN       NaN       NaN             male  
92        Hans Rosling       NaN       NaN       NaN             male

In [59]:
talk1speak.rename(columns={"Gender_handcheck": "talk_gender"}, inplace = True)

In [60]:
talk1speak.head()

public_url  \
Talk_ID                                                      
1        https://www.ted.com/talks/al_gore_on_averting_...   
7        https://www.ted.com/talks/david_pogue_says_sim...   
53       https://www.ted.com/talks/majora_carter_s_tale...   
66       https://www.ted.com/talks/ken_robinson_says_sc...   
92       https://www.ted.com/talks/hans_rosling_shows_t...   

                                headline  \
Talk_ID                                    
1            Averting the climate crisis   
7                       Simplicity sells   
53                   Greening the ghetto   
66           Do schools kill creativity?   
92       The best stats you've ever seen   

                                               description    event duration  \
Talk_ID                                                                        
1        With the same humor and humanity he exuded in ...  TED2006  0:16:17   
7        New York Times columnist David Pogue takes aim...  TED2006  0:21:26   
53       In an emotionally charged talk, MacArthur-winn...  TED2006  0:18:36   
66       Sir Ken Robinson makes an entertaining and pro...  TED2006  0:19:24   
92       You've never seen data presented like this. Wi...  TED2006  0:19:50   

        published                                               tags  \
Talk_ID                                                                
1         6/27/06  alternative energy,cars,global issues,climate ...   
7         6/27/06  simplicity,entertainment,interface design,soft...   
53        6/27/06  MacArthur grant,cities,green,activism,politics...   
66        6/27/06  children,teaching,creativity,parenting,culture...   
92        6/27/06  demo,Asia,global issues,visualizations,global ...   

            views                                               text  \
Talk_ID                                                                
1         3266733    Thank you so much, Chris. And it's truly a g...   
7         1702201    (Music: "The Sound of Silence," Simon & Garf...   
53        2000421    If you're here today — and I'm very happy th...   
66       51614087    Good morning. How are you?    (Laughter)    ...   
92       12662135    About 10 years ago, I took on the task to te...   

             speaker_1 speaker_2 speaker_3 speaker_4 talk_gender  
Talk_ID                                                           
1              Al Gore       NaN       NaN       NaN        male  
7          David Pogue       NaN       NaN       NaN        male  
53       Majora Carter       NaN       NaN       NaN      female  
66        Ken Robinson       NaN       NaN       NaN        male  
92        Hans Rosling       NaN       NaN       NaN        male

In [21]:
talk1speak.shape

(966, 14)

### Gendering talks with one speaker

Now with that talks with just one speaker each have been "gendered," we turn our attention to the talks with multiple speakers. If the multiple speakers for a talk have the same gender, then we label the talk with that gender. 

We start with those that have two speakers. We will add the gender for the first speaker and then for the second speaker. Our procedure is similar to the above process for just one speaker. 

In [66]:
print(talk2speak.shape)

talk2speak.head()

(25, 13)


public_url  \
Talk_ID                                                      
118      https://www.ted.com/talks/sergey_brin_and_larr...   
222      https://www.ted.com/talks/the_jill_and_julia_show   
224      https://www.ted.com/talks/roy_gould_and_curtis...   
246      https://www.ted.com/talks/tod_machover_and_dan...   
322      https://www.ted.com/talks/bruno_bowden_folds_w...   

                                            headline  \
Talk_ID                                                
118                            The genesis of Google   
222                          The Jill and Julia Show   
224             A preview of the WorldWide Telescope   
246      Inventing instruments that unlock new music   
322                      Blindfold origami and cello   

                                               description    event duration  \
Talk_ID                                                                        
118      Google co-founders Larry Page and Sergey Brin ...  TED2004  0:20:33   
222      Two TED favorites, Jill Sobule and Julia Sween...  TED2007  0:06:14   
224      Educator Roy Gould and researcher Curtis Wong ...  TED2008  0:06:42   
246      Tod Machover of MIT's Media Lab is devoted to ...  TED2008  0:20:41   
322      After Robert Lang's talk on origami at TED2008...  TED2008  0:02:58   

        published                                               tags    views  \
Talk_ID                                                                         
118        5/3/07  web,design,Google,culture,business,technology,...  1529641   
222       2/20/08  entertainment,comedy,humor,storytelling,collab...   507130   
224       2/27/08  telescopes,demo,astronomy,universe,science,tec...  1043036   
246       4/15/08  demo,entertainment,writing,live music,health c...   519734   
322        8/1/08                  origami,entertainment,cello,music   384129   

                                                      text     speaker_1  \
Talk_ID                                                                    
118        Sergey Brin: I want to discuss a question I ...   Sergey Brin   
222        ♫ Jill Sobule: At a conference in Monterey b...   Jill Sobule   
224        Roy Gould: Less than a year from now, the wo...     Roy Gould   
246        The first idea I'd like to suggest is that w...  Tod Machover   
322        Hello everyone. And so the two of us are her...  Bruno Bowden   

                speaker_2 speaker_3 speaker_4  
Talk_ID                                        
118            Larry Page       NaN       NaN  
222         Julia Sweeney       NaN       NaN  
224           Curtis Wong       NaN       NaN  
246            Dan Ellsey       NaN       NaN  
322      Rufus Cappadocia       NaN       NaN

We're going to make a copy of our talks with two speakers.

In [67]:
talk2temp = talk2speak.copy()

In [69]:
talk2temp.shape

(25, 13)

In [70]:
# Left merge to add the gender for the first speaker
talk2temp = talk2temp.reset_index().merge(gender_slice, left_on = "speaker_1", right_on = "speaker", how = "left")
talk2temp.set_index("Talk_ID", inplace = True)

# Drop the additional speaker column and rename the gender column 
talk2temp = talk2temp.drop(columns = ['speaker'])
talk2temp = talk2temp.rename(columns={"Gender_handcheck": "speaker_1_gender"})

talk2temp.head()

public_url  \
Talk_ID                                                      
118      https://www.ted.com/talks/sergey_brin_and_larr...   
222      https://www.ted.com/talks/the_jill_and_julia_show   
224      https://www.ted.com/talks/roy_gould_and_curtis...   
246      https://www.ted.com/talks/tod_machover_and_dan...   
322      https://www.ted.com/talks/bruno_bowden_folds_w...   

                                            headline  \
Talk_ID                                                
118                            The genesis of Google   
222                          The Jill and Julia Show   
224             A preview of the WorldWide Telescope   
246      Inventing instruments that unlock new music   
322                      Blindfold origami and cello   

                                               description    event duration  \
Talk_ID                                                                        
118      Google co-founders Larry Page and Sergey Brin ...  TED2004  0:20:33   
222      Two TED favorites, Jill Sobule and Julia Sween...  TED2007  0:06:14   
224      Educator Roy Gould and researcher Curtis Wong ...  TED2008  0:06:42   
246      Tod Machover of MIT's Media Lab is devoted to ...  TED2008  0:20:41   
322      After Robert Lang's talk on origami at TED2008...  TED2008  0:02:58   

        published                                               tags    views  \
Talk_ID                                                                         
118        5/3/07  web,design,Google,culture,business,technology,...  1529641   
222       2/20/08  entertainment,comedy,humor,storytelling,collab...   507130   
224       2/27/08  telescopes,demo,astronomy,universe,science,tec...  1043036   
246       4/15/08  demo,entertainment,writing,live music,health c...   519734   
322        8/1/08                  origami,entertainment,cello,music   384129   

                                                      text     speaker_1  \
Talk_ID                                                                    
118        Sergey Brin: I want to discuss a question I ...   Sergey Brin   
222        ♫ Jill Sobule: At a conference in Monterey b...   Jill Sobule   
224        Roy Gould: Less than a year from now, the wo...     Roy Gould   
246        The first idea I'd like to suggest is that w...  Tod Machover   
322        Hello everyone. And so the two of us are her...  Bruno Bowden   

                speaker_2 speaker_3 speaker_4 speaker_1_gender  
Talk_ID                                                         
118            Larry Page       NaN       NaN             male  
222         Julia Sweeney       NaN       NaN           female  
224           Curtis Wong       NaN       NaN             male  
246            Dan Ellsey       NaN       NaN             male  
322      Rufus Cappadocia       NaN       NaN             male

In [71]:
# Repeat the above process on the second speaker
# Left merge to add the gender for the second speaker
talk2temp = talk2temp.reset_index().merge(gender_slice, left_on = "speaker_2", right_on = "speaker", how = "left")
talk2temp.set_index("Talk_ID", inplace = True)

# Drop the additional speaker column and rename the gender column 
talk2temp = talk2temp.drop(columns = ['speaker'])
talk2temp = talk2temp.rename(columns={"Gender_handcheck": "speaker_2_gender"})

talk2temp.head()

public_url  \
Talk_ID                                                      
118      https://www.ted.com/talks/sergey_brin_and_larr...   
222      https://www.ted.com/talks/the_jill_and_julia_show   
224      https://www.ted.com/talks/roy_gould_and_curtis...   
246      https://www.ted.com/talks/tod_machover_and_dan...   
322      https://www.ted.com/talks/bruno_bowden_folds_w...   

                                            headline  \
Talk_ID                                                
118                            The genesis of Google   
222                          The Jill and Julia Show   
224             A preview of the WorldWide Telescope   
246      Inventing instruments that unlock new music   
322                      Blindfold origami and cello   

                                               description    event duration  \
Talk_ID                                                                        
118      Google co-founders Larry Page and Sergey Brin ...  TED2004  0:20:33   
222      Two TED favorites, Jill Sobule and Julia Sween...  TED2007  0:06:14   
224      Educator Roy Gould and researcher Curtis Wong ...  TED2008  0:06:42   
246      Tod Machover of MIT's Media Lab is devoted to ...  TED2008  0:20:41   
322      After Robert Lang's talk on origami at TED2008...  TED2008  0:02:58   

        published                                               tags    views  \
Talk_ID                                                                         
118        5/3/07  web,design,Google,culture,business,technology,...  1529641   
222       2/20/08  entertainment,comedy,humor,storytelling,collab...   507130   
224       2/27/08  telescopes,demo,astronomy,universe,science,tec...  1043036   
246       4/15/08  demo,entertainment,writing,live music,health c...   519734   
322        8/1/08                  origami,entertainment,cello,music   384129   

                                                      text     speaker_1  \
Talk_ID                                                                    
118        Sergey Brin: I want to discuss a question I ...   Sergey Brin   
222        ♫ Jill Sobule: At a conference in Monterey b...   Jill Sobule   
224        Roy Gould: Less than a year from now, the wo...     Roy Gould   
246        The first idea I'd like to suggest is that w...  Tod Machover   
322        Hello everyone. And so the two of us are her...  Bruno Bowden   

                speaker_2 speaker_3 speaker_4 speaker_1_gender  \
Talk_ID                                                          
118            Larry Page       NaN       NaN             male   
222         Julia Sweeney       NaN       NaN           female   
224           Curtis Wong       NaN       NaN             male   
246            Dan Ellsey       NaN       NaN             male   
322      Rufus Cappadocia       NaN       NaN             male   

        speaker_2_gender  
Talk_ID                   
118                 male  
222               female  
224                 male  
246                 male  
322                 male

Now we check which rows have the same genders for both speakers:

In [50]:
talk2speak[talk2speak["speaker_1_gender"] == talk2speak["speaker_2_gender"]].shape

(18, 15)

In [51]:
talk2speak[talk2speak["speaker_1_gender"] != talk2speak["speaker_2_gender"]].shape

(7, 15)